components/Chat.tsx

```diff
@@ -100,51 +107,136 @@ export default function Chat() {
         Chat
       </Typography>
       <Box
-        overflow="scroll"
         border="2px solid #D6D6D6"
         borderRadius="16px"
+        height="80vh"
         display="flex"
         flexDirection="column"
-        gap="2rem"
-        padding="2rem"
-        textAlign="start"
       >
-        {messages.map((message) => {
-          const isUser = message.role === "user";
-          return (
-            <Box
-              display="flex"
-              flexDirection={isUser ? "row-reverse" : "row"}
-              gap="1rem"
-              alignItems="flex-end"
-            >
-              {isUser && (
-                <img
-                  src={pfp}
-                  alt="profile"
-                  width={40}
-                  height={40}
-                  style={{ borderRadius: "50%" }}
-                />
-              )}
-              <Box
-                padding="1rem"
-                bgcolor={isUser ? "#1E87F0" : "grey.200"}
-                color={isUser ? "white" : "#383838"}
-                borderRadius={isUser ? "1rem 1rem 0 1rem" : "1rem 1rem 1rem 0"}
-                maxWidth="60%"
-              >
-                {message.content.trim()}
-              </Box>
-            </Box>
-          );
-        })}
-        {/* TODO: make it look good */}
-        <form onSubmit={handleSubmit}>
-          <Input value={query} onChange={(e) => setQuery(e.target.value)} />
-          <Button type="submit" disabled={isLoading}>Send</Button>
-        </form>
+        <Box
+          overflow="scroll"
+          height="100%"
+          display="flex"
+          flexDirection="column"
+          gap="2rem"
+          padding="2rem"
+          textAlign="start"
+        >
+          {messages.map((message) => (
+            <React.Fragment key={message.date}>
+              <ChatBubble message={message} userPFP={pfp} />
+            </React.Fragment>
+          ))}
+        </Box>
+        {isLoading && <ChatSystemLoadingBubble />}
+        <ChatInput
+          handleSubmit={handleSubmit}
+          value={query}
+          onChange={(e) => setQuery(e.target.value)}
+          isLoading={isLoading}
+        />
+      </Box>
+    </Box>
```

Comment:
![image](https://github.com/user-attachments/assets/287b99bc-c464-4555-bab4-3c8cb5e81358)
Not sure if this is intended, but the scrollbar for horizontal seems to be above the loading chat bubble. 

Is there any chance we can make it look something like the picture below? (My browser is chromium on ubuntu, not sure if that is affecting appearance)

![image](https://github.com/user-attachments/assets/3669dc9e-1fd3-402e-b982-00f87f74b1c5)

Requirements:
- Typing text bubble for the system is displayed underneath the latest system message
- use the triple dots for typing indication

Reply:
Not sure how time consuming this would be. If you're busy with other assignments we could just keep the spinner , but i think we should still move the loading text bubble to be underneath the last sent message

Reply:
Also is it possible to remove the horizontal scroll bar ?

Reply:
Thanks for the suggestion, I fixed the typing text bubble so it should show right below the latest message now.
I also added an animated triple-dot loading indicator like the one you pictured.
I can't reproduce the horizontal scroll bar but I believe it should be fixed by `box-sizing: border-box` for the chat box which has a thin border, maybe that's causing tiny amount of horizontal scroll. Let me know if it resolves the issue